Sage/carbon
# fix(icon-button): fix background color when component is disabled (FE-4867)

A bug was introducted in v101.2.0 where the background color of a disabled icon-button was made to
be grey. The correct colour should be transparent.

fixes #4742

### Proposed behaviour

![Screenshot 2022-02-04 at 13 42 48](https://user-images.githubusercontent.com/56251247/152538888-7ec317f7-ec67-453b-a5cd-de90eabb592a.png)

![Screenshot 2022-02-04 at 13 43 27](https://user-images.githubusercontent.com/56251247/152538969-f6e7aa4f-53be-4229-bfd4-9779d5ad6531.png)

### Current behaviour

![Screenshot 2022-02-04 at 13 44 12](https://user-images.githubusercontent.com/56251247/152539102-bcd994d6-5ae2-41f8-8a23-71f265bb3803.png)

### Checklist




#### QA


### Additional context

Issue was raised against `Confirm` but after some investigation I found that the root source of this issue was actually with `icon-button`.

### Testing instructions



- Check `Confirm -> disableCancel` story to make sure that the `X` has the correct styling.
- New story for `icon-button` has been created called `disabled` so that Chromatic can catch any visual changes in future.
